server/src/controllers/auth.js

```diff
@@ -88,8 +88,12 @@ module.exports = {
 					ticket: userTicket
 				}
 			}).then((sres) => {
-				if (sres.data.response.params.result === 'OK') {
-					if (idToVerify === sres.data.response.params.steamid) {
+				const params = sres.data.response.params;
+				if (!params)
+					res.sendStatus(400); // Bad request, it'll have an error object instead
```

Comment:
I think it is safe to respond with the error reasons by simply passing the error:

Maybe if there is a silent ban or something maybe not, needs @Gocnak 's opinion

```node
res.sendStatus(400).json(sres.data.response.error);
```

It could be prettied into a renamed format, but I think it is fine

Reply:
As per the docs, [`sendStatus(status)`](https://expressjs.com/en/4x/api.html#res.sendStatus) does internal response handling, you would instead need to do
```
res.status(400).json(sres.data.response.error);
```

As per the guide on [`response.json()`](https://expressjs.com/en/4x/api.html#res.json)

Reply:
Are we happy to respond to clients with the steam error message though? Would it let them know of hidden bans or something like that?

Reply:
Yes good point. Let's just make this an error 400 here for now, as the code originally had.